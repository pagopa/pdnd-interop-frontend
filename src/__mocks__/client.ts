import { Client } from '../../types'

export const clientActiveAgreementActiveEservicePublished: Client = {
  id: 'dsfsdjf-sdfjsdfj-sdfjdsf',
  name: 'dipendenti del comune',
  description: 'il client dei dipendenti del mio comune',
  state: 'ACTIVE',
  agreement: {
    id: 'sjfaisds-sdfjsaodfj-sfajd',
    state: 'ACTIVE',
    descriptor: {
      id: 'djfiosj-dfjdsofj-dfjsdf',
      state: 'PUBLISHED',
      version: '5',
    },
  },
  eservice: {
    id: 'dsnodsi-sdfdsljf-sdjfodsjf',
    name: 'anagrafe comune di roma',
    provider: {
      institutionId: 'abc_fd_df',
      description: 'Comune di Roma',
    },
  },
  purposes: 'accesso pieno',
}

export const clientActiveAgreementActiveEserviceDeprecated: Client = {
  id: 'dsfsdjf-sdfjsdfj-sdfjdsf',
  name: 'dipendenti del comune',
  description: 'il client dei dipendenti del mio comune',
  state: 'ACTIVE',
  agreement: {
    id: 'sjfaisds-sdfjsaodfj-sfajd',
    state: 'ACTIVE',
    descriptor: {
      id: 'djfiosj-dfjdsofj-dfjsdf',
      state: 'DEPRECATED',
      version: '5',
    },
  },
  eservice: {
    id: 'dsnodsi-sdfdsljf-sdjfodsjf',
    name: 'anagrafe comune di roma',
    provider: {
      institutionId: 'abc_fd_df',
      description: 'Comune di Roma',
    },
    activeDescriptor: {
      id: 'sdjfsdjf-djfsdj-sdjfdsj',
      state: 'PUBLISHED',
      version: '7',
    },
  },
  purposes: 'accesso pieno',
}

export const clientSuspendedAgreementActiveEservicePublished: Client = {
  id: 'dsfsdjf-sdfjsdfj-sdfjdsf',
  name: 'dipendenti del comune',
  description: 'il client dei dipendenti del mio comune',
  state: 'SUSPENDED',
  agreement: {
    id: 'sjfaisds-sdfjsaodfj-sfajd',
    state: 'ACTIVE',
    descriptor: {
      id: 'djfiosj-dfjdsofj-dfjsdf',
      state: 'PUBLISHED',
      version: '5',
    },
  },
  eservice: {
    id: 'dsnodsi-sdfdsljf-sdjfodsjf',
    name: 'anagrafe comune di roma',
    provider: {
      institutionId: 'abc_fd_df',
      description: 'Comune di Roma',
    },
  },
  purposes: 'accesso pieno',
}

export const clientActiveAgreementSuspendedEservicePublished: Client = {
  id: 'dsfsdjf-sdfjsdfj-sdfjdsf',
  name: 'dipendenti del comune',
  description: 'il client dei dipendenti del mio comune',
  state: 'ACTIVE',
  agreement: {
    id: 'sjfaisds-sdfjsaodfj-sfajd',
    state: 'SUSPENDED',
    descriptor: {
      id: 'djfiosj-dfjdsofj-dfjsdf',
      state: 'PUBLISHED',
      version: '5',
    },
  },
  eservice: {
    id: 'dsnodsi-sdfdsljf-sdjfodsjf',
    name: 'anagrafe comune di roma',
    provider: {
      institutionId: 'abc_fd_df',
      description: 'Comune di Roma',
    },
  },
  purposes: 'accesso pieno',
}

export const clientActiveAgreementActiveEserviceSuspended: Client = {
  id: 'dsfsdjf-sdfjsdfj-sdfjdsf',
  name: 'dipendenti del comune',
  description: 'il client dei dipendenti del mio comune',
  state: 'ACTIVE',
  agreement: {
    id: 'sjfaisds-sdfjsaodfj-sfajd',
    state: 'ACTIVE',
    descriptor: {
      id: 'djfiosj-dfjdsofj-dfjsdf',
      state: 'SUSPENDED',
      version: '5',
    },
  },
  eservice: {
    id: 'dsnodsi-sdfdsljf-sdjfodsjf',
    name: 'anagrafe comune di roma',
    provider: {
      institutionId: 'abc_fd_df',
      description: 'Comune di Roma',
    },
  },
  purposes: 'accesso pieno',
}
