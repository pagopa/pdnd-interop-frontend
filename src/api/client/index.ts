export * from './client.queries'
export * from './client.mutations'
export * from './client.services'
export * from './client.downloads'
