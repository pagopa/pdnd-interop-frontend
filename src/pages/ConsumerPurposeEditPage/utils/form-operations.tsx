import {
  CheckboxGroup,
  RadioGroup,
  Select,
  TextField,
} from '@/components/shared/ReactHookFormInputs'
import type { InputOption } from '@/types/common.types'
import React from 'react'
import identity from 'lodash/identity'
import type {
  Dependency,
  DynamicFormOperations,
  Questions,
  QuestionV1,
  QuestionV2,
} from '../types/risk-analysis.types'
import { RiskAnalysisSwitch } from '../components/PurposeEditStep2RiskAnalysis/RiskAnalysisSwitch'

export const dynamicFormOperationsVersions: DynamicFormOperations = {
  '1.0': {
    getUpdatedQuestions: (values, riskAnalysis) => {
      const updatedQuestions = riskAnalysis.questions.reduce((acc, next) => {
        // We are sure we can only queue the questions that come after the current one
        // because the dependencies are always on the questions that come before.
        // E.g. question 3 can only have question 1 and/or 2 as a dependency,
        // not question 4 and subsequent
        const queuedQuestions = Object.values(acc).map((d) => d.id)

        // Check the dependencies of this questions
        const satisfiesAllDependencies = next.dependencies.every((d) => {
          // If the question is a dependency
          const isInArray = queuedQuestions.includes(d.id)
          const id = d.id as string

          // and if the field currently has the same value as stated in the dependency
          const hasSameValue = Array.isArray(values[id])
            ? (values[id] as Array<unknown>).includes(d.value)
            : values[id] === d.value

          return isInArray && hasSameValue
        })

        // Then it needs to be added as a new question to ask
        if (satisfiesAllDependencies) {
          acc[next.id] = next
        }

        return acc
      }, {} as Questions)
      return updatedQuestions
    },

    buildForm: (questions, _, lang, t) => {
      const questionKeys = Object.keys(questions)

      const formComponents = questionKeys.map((id, i) => {
        const { type, label, options, infoLabel } = questions[id] as QuestionV1

        const sx = questionKeys.length - 1 === i ? { mb: 0 } : {}
        const inputOptions = options ? options.map((o) => ({ ...o, label: o.label[lang] })) : []
        const commonProps = {
          key: id,
          name: id,
          label: label[lang],
          infoLabel: infoLabel ? infoLabel[lang] : undefined,
          sx,
        }

        switch (type) {
          case 'text':
            return <TextField {...commonProps} rules={{ required: true }} />

          case 'select-one':
            return (
              <Select
                {...commonProps}
                options={inputOptions}
                emptyLabel={t('edit.step2.emptyLabel')}
                rules={{ required: true }}
              />
            )

          case 'checkbox':
            const errorMessage =
              options && options?.length > 1
                ? t('edit.step2.multiCheckboxField.validation.mixed.required')
                : t('edit.step2.singleCheckboxField.validation.mixed.required')
            return (
              <CheckboxGroup
                {...commonProps}
                options={inputOptions}
                rules={{
                  validate: (value) =>
                    (typeof value !== 'undefined' && value.length > 0) || errorMessage,
                }}
              />
            )

          case 'radio':
            return <RadioGroup {...commonProps} options={inputOptions} rules={{ required: true }} />
        }
      })

      return { formComponents: formComponents, isSubmitBtnDisabled: false }
    },
  },
  '2.0': {
    getUpdatedQuestions: (values, riskAnalysis) => {
      const questions = riskAnalysis.questions as Array<QuestionV2>

      // Filters all the questions that not satisfies the dependency inside the "dependencies" property
      const updatedQuestions = questions.filter(({ dependencies }) => {
        function satisfiesDependency(dep: QuestionV2['dependencies'][0]) {
          if (Array.isArray(values[dep.id])) {
            return (values[dep.id] as Array<unknown>).includes(dep.value)
          }
          if (Array.isArray(dep.value)) {
            return dep.value.includes(values[dep.id])
          }
          return values[dep.id] === dep.value
        }
        // If length of dependencies array is 0 the "every" method returns true
        return dependencies.every(satisfiesDependency)
      })

      // Array<Question> -> Record<"id", Question>
      return updatedQuestions.reduce((acc, next) => ({ ...acc, [next.id]: next }), {} as Questions)
    },

    buildForm: (questions, formMethods, lang, t) => {
      const values = formMethods.getValues()

      const formComponents: Array<React.ReactNode> = []

      const questionIds = Object.keys(questions)

      type QuestionOption = Exclude<QuestionV2['options'], undefined>[0]

      function checkOptionDependency(dependency: Dependency) {
        const currentDepValue = values[dependency.id]

        const hasDependency =
          currentDepValue === dependency.value ||
          (Array.isArray(currentDepValue) &&
            (currentDepValue as string[]).includes(dependency.value as string))

        return hasDependency
      }

      function parseOption(option: QuestionOption, { hideOption, id, defaultValue }: QuestionV2) {
        // if the key "hideOption" is present in the question object and the conditions are satisfied
        // the option will be not added to the array of options
        const shouldHideOption =
          hideOption &&
          hideOption[option.value] &&
          hideOption[option.value].some(checkOptionDependency)

        if (!shouldHideOption) {
          return { value: option.value, label: option.label[lang] }
        }

        if ((values[id] as string[])?.includes(option.value)) {
          formMethods.setValue(id, defaultValue)
        }
      }

      function buildFormQuestionComponents(
        { id, type, ...question }: QuestionV2,
        inputOptions: Array<InputOption>,
        isLast: boolean
      ) {
        const questionComponents: Array<React.ReactNode> = []

        const maxLength = question?.validation?.maxLength
        const isRequired = question.required
        const isMultipleChoice = question.dataType === 'multi'

        let label = question.label[lang]
        let infoLabel = question.infoLabel && question.infoLabel[lang]

        if (maxLength) {
          const maxLengthLabel = t('edit.step2.validation.maxLength', { num: maxLength })
          if (infoLabel) {
            infoLabel += `. ${maxLengthLabel}`
          } else {
            infoLabel = maxLengthLabel
          }
        }

        const labelValidation: Array<string> = []

        if (isRequired) {
          labelValidation.push(t('edit.step2.validation.required'))
        }

        if (isMultipleChoice) {
          labelValidation.push(t('edit.step2.validation.multipleChoice'))
        }

        if (labelValidation.length > 0) {
          label += ` (${labelValidation.join(', ')})`
        }

        const sx = isLast ? { mb: 0 } : {}
        const commonProps = { key: id, name: id, label, infoLabel, sx }

        switch (type) {
          case 'text':
            questionComponents.push(
              <TextField {...commonProps} inputProps={{ maxLength }} rules={{ required: true }} />
            )
            break
          case 'select-one':
            questionComponents.push(
              <Select
                {...commonProps}
                options={inputOptions}
                emptyLabel={t('edit.step2.emptyLabel')}
                rules={{ required: true }}
              />
            )
            break
          case 'checkbox':
            questionComponents.push(
              <CheckboxGroup
                {...commonProps}
                options={inputOptions}
                rules={{
                  validate: (value) =>
                    (typeof value !== 'undefined' && value.length > 0) ||
                    t('edit.step2.multiCheckboxField.validation.mixed.required'),
                }}
              />
            )
            break
          case 'radio':
            questionComponents.push(
              <RadioGroup {...commonProps} options={inputOptions} rules={{ required: true }} />
            )
            break
          case 'switch':
            questionComponents.push(
              <RiskAnalysisSwitch
                {...commonProps}
                options={inputOptions}
                rules={{
                  validate: (value) =>
                    (typeof value === 'boolean' && value === true) ||
                    t('edit.step2.riskAnalysisSwitch.validation.boolean.isValue'),
                }}
              />
            )
            break
        }

        return questionComponents
      }

      for (let i = 0; i < questionIds.length; i++) {
        const questionId = questionIds[i]

        const question = questions[questionId] as QuestionV2

        const inputOptions = (question.options
          ?.map((option) => parseOption(option, question))
          .filter(identity) || []) as Array<InputOption>

        formComponents.push(
          ...buildFormQuestionComponents(question, inputOptions, questionIds.length - 1 === i)
        )
      }

      return { formComponents, isSubmitBtnDisabled: false }
    },
  },
}

export function getFormOperations(version: string) {
  const dynamicFormOperations = dynamicFormOperationsVersions[version]

  if (!dynamicFormOperations) {
    throw new Error('There is no business logic for this specific risk analysis data form')
  }

  return dynamicFormOperations
}
