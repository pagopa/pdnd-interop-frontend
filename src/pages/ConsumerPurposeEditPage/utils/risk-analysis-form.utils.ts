import type { Dependency, FormConfigQuestion, HideOption } from '@/api/api.generatedTypes'
import type { AnswerValue, Answers, Questions } from '../types/risk-analysis-form.types'

/**
 * Transform the answers from the backend to the frontend format
 * The answer from the backend arrives as an array of strings, but the frontend
 * needs a specific format depending on the `visualType` value.
 *
 * - `switch` needs a boolean
 * - `checkbox` needs an array of strings
 * - the rest needs a string
 *
 * @param backendAnswer The answer from the backend
 * @param question The question that the answer belongs to
 * @returns The answer in the type that the frontend needs
 */
export function getFrontendAnswerValue(
  backendAnswer: Array<string>,
  visualType?: FormConfigQuestion['visualType']
): AnswerValue {
  if (visualType === 'switch') {
    return backendAnswer[0] === 'true'
  }

  if (visualType === 'checkbox') {
    return backendAnswer
  }

  return backendAnswer[0]
}

/**
 * Transform the answers from the frontend to the backend format
 * The backend always needs an array of strings.
 *
 * @param answer The answer from the frontend
 * @returns The answer in the type that the backend needs
 * */
export function getBackendAnswerValue(answer: AnswerValue): Array<string> {
  if (typeof answer === 'boolean') {
    return [answer.toString()]
  }

  if (Array.isArray(answer)) {
    return answer
  }

  if (typeof answer === 'string') {
    return [answer]
  }

  throw new Error(`Invalid answer type: ${typeof answer} \n\n ${JSON.stringify(answer, null, 2)}`)
}

/**
 * Returns the valid answers that should be sent to the backend.
 * The answers that should be sent to the backend are only the answers
 * to the questions that are currently visible to the user.
 * The answers to the questions that are not visible to the user are not sent.
 *
 * @param currentQuestionsIds - the ids of the questions that are currently visible to the user
 * @param answers - the answers to all the questions
 * @returns the valid answers that should be sent to the backend
 * */
export function getValidAnswers(currentQuestionsIds: Array<string>, answers: Answers) {
  return Object.keys(answers).reduce<Record<string, Array<string>>>((acc, key) => {
    // If the question is actually visible, add it to the valid answers
    if (currentQuestionsIds.includes(key)) {
      return { ...acc, [key]: getBackendAnswerValue(answers[key]) }
    }
    return acc
  }, {})
}

/**
 * Returns true if the dependency is satisfied, false otherwise.
 *
 * @param dependency - the dependency
 * @param answers - the actual form values
 * @returns `true` if the dependency is satisfied, `false` otherwise
 * */
export function isDependencySatisfied(dependency: Dependency | HideOption, answers: Answers) {
  const answer = answers[dependency.id]
  if (Array.isArray(answer)) {
    return answer.includes(dependency.value)
  }

  return answer === dependency.value
}

/**
 * Returns the updated question data.
 *
 * @param answers - the actual form values
 * @param riskAnalysis - the risk analysis document
 *
 * @returns the updated questions data
 * */
export function getUpdatedQuestions(
  answers: Answers,
  riskAnalysisQuestions: FormConfigQuestion[]
): Questions {
  // Filters all the questions that not satisfies the dependency inside the "dependencies" property
  return riskAnalysisQuestions.reduce<Questions>((acc, question) => {
    const doesSatisfyDeps = question.dependencies.every((dependency) =>
      isDependencySatisfied(dependency, answers)
    )
    if (doesSatisfyDeps) {
      acc[question.id] = question
    }
    return acc
  }, {})
}

/**
 * Returns the default values for the risk analysis form.
 * The default values are the values that the form should have when it is first loaded.
 * The default values are the values that the user has already answered, if any.
 * If the user has not answered any question, the values are the default values
 * defined in the risk analysis document.
 *
 * @param riskAnalysisConfig the risk analysis document
 * @param backendAnswers the answers data, taken from the Purpose
 * @returns the default values for the risk analysis form
 */
export function getRiskAnalysisDefaultValues(
  riskAnalysisConfigQuestions: FormConfigQuestion[],
  backendAnswers?: Record<string, Array<string>>
): Answers {
  return riskAnalysisConfigQuestions.reduce<Answers>((acc, question) => {
    const answer = backendAnswers?.[question.id] ?? question.defaultValue
    acc[question.id] = getFrontendAnswerValue(answer, question.visualType)

    if (!acc[question.id] && question.dataType === 'FREETEXT') {
      acc[question.id] = ''
    }

    return acc
  }, {})
}
