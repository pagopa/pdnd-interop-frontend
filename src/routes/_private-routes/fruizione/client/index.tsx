import React from 'react'
import { createFileRoute } from '@tanstack/react-router'

export const Route = createFileRoute('/_private-routes/fruizione/client/')({
  staticData: {
    authLevels: ['admin', 'support', 'security'],
    routeKey: 'SUBSCRIBE_CLIENT_LIST',
  },
  component: () => (
    <div>
      Hello /_authentication-guard/_tos-guard/_app-layout/_authorization-guard/fruizione/client/!
    </div>
  ),
})
