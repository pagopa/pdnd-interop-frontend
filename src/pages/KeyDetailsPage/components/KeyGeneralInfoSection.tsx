import { ClientQueries } from '@/api/client'
import { SectionContainer, SectionContainerSkeleton } from '@/components/layout/containers'
import { formatDateString } from '@/utils/format.utils'
import { Alert, Grid } from '@mui/material'
import { Stack } from '@mui/system'
import { InformationContainer } from '@pagopa/interop-fe-commons'
import React from 'react'
import { useTranslation } from 'react-i18next'

type KeyGeneralInfoSectionProps = {
  clientId: string
  kid: string
}

export const KeyGeneralInfoSection: React.FC<KeyGeneralInfoSectionProps> = ({ clientId, kid }) => {
  const { t } = useTranslation('key', { keyPrefix: 'edit.generalInformations' })
  const { data: publicKey } = ClientQueries.useGetSingleKey(clientId, kid)

  return (
    <Grid container>
      <Grid item xs={7}>
        <SectionContainer title={t('title')}>
          <Stack spacing={2}>
            <InformationContainer
              label={t('creationDateField.label')}
              content={publicKey ? formatDateString(publicKey.createdAt) : ''}
            />
            <InformationContainer
              label={t('uploaderField.label')}
              content={`${publicKey?.operator.name ?? ''} ${publicKey?.operator.familyName ?? ''}`}
            />
            <InformationContainer
              label={t('kidField.label')}
              content={publicKey?.keyId ?? ''}
              copyToClipboard={
                publicKey?.keyId
                  ? {
                      value: publicKey.keyId,
                      tooltipTitle: t('kidField.copySuccessFeedbackText'),
                    }
                  : undefined
              }
            />
            <InformationContainer
              label={t('clientIdField.label')}
              content={clientId}
              copyToClipboard={{
                value: clientId,
                tooltipTitle: t('clientIdField.copySuccessFeedbackText'),
              }}
            />
            {publicKey?.isOrphan && (
              <Alert severity="info">{t('operatorDeletedAlertMessage')}</Alert>
            )}
          </Stack>
        </SectionContainer>
      </Grid>
    </Grid>
  )
}

export const KeyGeneralInfoSectionSkeleton = () => {
  return (
    <Grid container>
      <Grid item xs={7}>
        <SectionContainerSkeleton height={227} />
      </Grid>
    </Grid>
  )
}
