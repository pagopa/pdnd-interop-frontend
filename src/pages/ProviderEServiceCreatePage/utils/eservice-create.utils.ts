import type {
  AgreementApprovalPolicy,
  DescriptorAttributes,
  ProducerEServiceDescriptor,
} from '@/api/api.generatedTypes'
import type { RemappedEServiceAttributes } from '@/types/attribute.types'
import { getKeys } from '@/utils/array.utils'
import { remapEServiceAttributes } from '@/utils/attribute.utils'
import isEqual from 'lodash/isEqual'

type DescriptorDataStep2ToCompare = {
  voucherLifespan: number
  audience: string[]
  agreementApprovalPolicy: AgreementApprovalPolicy
  version: string
  description: string
  dailyCallsPerConsumer: number
  dailyCallsTotal: number
}

export function compareDescriptorsStep2DataIfEquals(
  newDescriptorData: DescriptorDataStep2ToCompare,
  descriptorToCompare: ProducerEServiceDescriptor
) {
  const descriptorDataToCompare: DescriptorDataStep2ToCompare = {
    voucherLifespan: descriptorToCompare.voucherLifespan,
    audience: descriptorToCompare.audience,
    agreementApprovalPolicy: descriptorToCompare.agreementApprovalPolicy,
    version: descriptorToCompare.version,
    description: descriptorToCompare.description ?? '',
    dailyCallsPerConsumer: descriptorToCompare.dailyCallsPerConsumer,
    dailyCallsTotal: descriptorToCompare.dailyCallsTotal,
  }

  return getKeys(newDescriptorData).every((key) =>
    isEqual(newDescriptorData[key], descriptorDataToCompare[key])
  )
}

export function compareAttributesStep3IfEquals(
  newAttributes: RemappedEServiceAttributes,
  attributesToCompare: DescriptorAttributes
) {
  const remappedAttributesToCompare = remapEServiceAttributes(attributesToCompare)

  const newAttributesToCompare = {
    certified: newAttributes.certified.filter((group) => group.attributes.length > 0),
    verified: newAttributes.verified.filter((group) => group.attributes.length > 0),
    declared: newAttributes.declared.filter((group) => group.attributes.length > 0),
  }

  return getKeys(newAttributes).every((key) =>
    isEqual(newAttributesToCompare[key], remappedAttributesToCompare[key])
  )
}
