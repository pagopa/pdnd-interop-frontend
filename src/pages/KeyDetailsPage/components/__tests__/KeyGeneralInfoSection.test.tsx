import React from 'react'
import { renderWithApplicationContext } from '@/utils/testing.utils'
import { KeyGeneralInfoSection, KeyGeneralInfoSectionSkeleton } from '../KeyGeneralInfoSection'
import { vi } from 'vitest'
import { ClientQueries } from '@/api/client'
import { createMockPublicKey } from '../../../../../__mocks__/data/key.mocks'
import type { PublicKey } from '@/api/api.generatedTypes'

const mockUseGetSingleKey = (publicKey: PublicKey | undefined) => {
  vi.spyOn(ClientQueries, 'useGetSingleKey').mockReturnValue({
    data: publicKey,
  } as unknown as ReturnType<typeof ClientQueries.useGetSingleKey>)
}

describe('KeyGeneralInfoSection', () => {
  it('should match the snapshot', () => {
    mockUseGetSingleKey(createMockPublicKey())
    const { baseElement } = renderWithApplicationContext(
      <KeyGeneralInfoSection clientId="clientId" kid="kid" />,
      {
        withReactQueryContext: true,
      }
    )
    expect(baseElement).toMatchSnapshot()
  })

  it('should not render if the public key is undefined', () => {
    mockUseGetSingleKey(undefined)
    const { container } = renderWithApplicationContext(
      <KeyGeneralInfoSection clientId="clientId" kid="kid" />,
      {
        withReactQueryContext: true,
      }
    )
    expect(container).toBeEmptyDOMElement()
  })
})

describe('KeyGeneralInfoSectionSkeleton', () => {
  it('should match the snapshot', () => {
    const { baseElement } = renderWithApplicationContext(<KeyGeneralInfoSectionSkeleton />, {
      withReactQueryContext: true,
    })
    expect(baseElement).toMatchSnapshot()
  })
})
