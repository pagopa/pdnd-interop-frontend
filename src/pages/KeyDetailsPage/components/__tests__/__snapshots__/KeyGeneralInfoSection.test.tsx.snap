// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`KeyGeneralInfoSection > should match the snapshot 1`] = `
<body>
  <div>
    <div
      class="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-7 css-uqwprf-MuiGrid-root"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1w5x330-MuiPaper-root"
        >
          <div
            class="css-nen11g-MuiStack-root"
          >
            <h2
              class="MuiTypography-root MuiTypography-overline css-b3fch5-MuiTypography-root"
            >
              title
            </h2>
          </div>
          <div
            class="MuiBox-root css-1yuhvjn"
          >
            <div
              class="MuiStack-root css-1p5q5e5-MuiStack-root"
            >
              <div
                class="css-19qjufr-MuiStack-root"
              >
                <div
                  class="MuiBox-root css-iuqs4u"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
                  >
                    creationDateField.label
                  </p>
                </div>
                <div
                  class="MuiBox-root css-1rr4qq7"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body2 css-hoiwr4-MuiTypography-root"
                  >
                    28 febbraio 2023
                  </p>
                </div>
              </div>
              <div
                class="css-19qjufr-MuiStack-root"
              >
                <div
                  class="MuiBox-root css-iuqs4u"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
                  >
                    uploaderField.label
                  </p>
                </div>
                <div
                  class="MuiBox-root css-1rr4qq7"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body2 css-hoiwr4-MuiTypography-root"
                  >
                    Mario Rossi
                  </p>
                </div>
              </div>
              <div
                class="css-19qjufr-MuiStack-root"
              >
                <div
                  class="MuiBox-root css-iuqs4u"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
                  >
                    kidField.label
                  </p>
                </div>
                <div
                  class="MuiBox-root css-1rr4qq7"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body2 css-hoiwr4-MuiTypography-root"
                  >
                    Fxod41P3BZoe2HT6BuEw3SNFlu9ufAkYgpmtBoNuVkg
                    <button
                      aria-label="Copia"
                      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-vqczhe-MuiButtonBase-root-MuiIconButton-root"
                      data-mui-internal-clone-element="true"
                      role="button"
                      tabindex="0"
                      type="button"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
                        data-testid="ContentCopyIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M16 1H4c-1.1 0-2 .9-2 2v14h2V3h12V1zm3 4H8c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h11c1.1 0 2-.9 2-2V7c0-1.1-.9-2-2-2zm0 16H8V7h11v14z"
                        />
                      </svg>
                      <span
                        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                      />
                    </button>
                  </p>
                </div>
              </div>
              <div
                class="css-19qjufr-MuiStack-root"
              >
                <div
                  class="MuiBox-root css-iuqs4u"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
                  >
                    clientIdField.label
                  </p>
                </div>
                <div
                  class="MuiBox-root css-1rr4qq7"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body2 css-hoiwr4-MuiTypography-root"
                  >
                    clientId
                    <button
                      aria-label="Copia"
                      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-vqczhe-MuiButtonBase-root-MuiIconButton-root"
                      data-mui-internal-clone-element="true"
                      role="button"
                      tabindex="0"
                      type="button"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
                        data-testid="ContentCopyIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M16 1H4c-1.1 0-2 .9-2 2v14h2V3h12V1zm3 4H8c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h11c1.1 0 2-.9 2-2V7c0-1.1-.9-2-2-2zm0 16H8V7h11v14z"
                        />
                      </svg>
                      <span
                        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                      />
                    </button>
                  </p>
                </div>
              </div>
              <div
                class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiAlert-root MuiAlert-standardInfo MuiAlert-standard css-1fuz9qr-MuiPaper-root-MuiAlert-root"
                role="alert"
              >
                <div
                  class="MuiAlert-icon css-1ytlwq5-MuiAlert-icon"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                    data-testid="InfoOutlinedIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M11,9H13V7H11M12,20C7.59,20 4,16.41 4,12C4,7.59 7.59,4 12,4C16.41,4 20,7.59 20, 12C20,16.41 16.41,20 12,20M12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0,0 22,12A10, 10 0 0,0 12,2M11,17H13V11H11V17Z"
                    />
                  </svg>
                </div>
                <div
                  class="MuiAlert-message css-1pxa9xg-MuiAlert-message"
                >
                  operatorDeletedAlertMessage
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`KeyGeneralInfoSectionSkeleton > should match the snapshot 1`] = `
<body>
  <div>
    <div
      class="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-7 css-uqwprf-MuiGrid-root"
      >
        <span
          class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-4i8vzz-MuiSkeleton-root"
          style="height: 227px;"
        />
      </div>
    </div>
  </div>
</body>
`;
